src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -67,13 +67,17 @@ export class TableOfContents implements OnInit {
   }
 
   ngOnInit(): void {
-    this._scrollContainer = this.container ?
-      this._document.querySelectorAll(this.container)[0] : window;
-
-    this._scrollSubscription = Observable
-      .fromEvent(this._scrollContainer, 'scroll')
-      .debounceTime(10)
-      .subscribe(() => this.onScroll());
+    // On init, the sidenav content element doesn't yet exist, so it's not possible
+    // to subscribe to its scroll event until next tick (when it does exist).
+    Promise.resolve().then(() => {
```

Comment:
Couldn't you do this in one of the other lifecycle hooks?

Reply:
Fair point. I can move to `ngAfterViewInit`.

Reply:
Edit: this appears not to work.